app/controllers/stripe-setup/responsible-person/post.controller.js

```diff
@@ -2,27 +2,98 @@
+    validator: validateMandatoryField,
+    maxLength: 100
+  },
+  {
+    field: LAST_NAME_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 100
+  },
+  {
+    field: HOME_ADDRESS_LINE1_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 200
+  },
+  {
+    field: HOME_ADDRESS_LINE2_FIELD,
+    validator: validateOptionalField,
+    maxLength: 200
+  },
+  {
+    field: HOME_ADDRESS_CITY_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 100
+  },
+  {
+    field: HOME_ADDRESS_POSTCODE_FIELD,
+    validator: validatePostcode
+  }
+]
 
 module.exports = (req, res) => {
+  const errors = validationRules.reduce((errors, validationRule) => {
+    const errorMessage = validate(req, validationRule.field, validationRule.validator, validationRule.maxLength)
+    if (errorMessage) {
+      errors[validationRule.field] = errorMessage
+    }
+    return errors
+  }, {})
+
   const pageData = {
-    firstName: lodash.get(req, 'body.first-name', ''),
-    lastName: lodash.get(req, 'body.last-name', ''),
-    homeAddressLine1: lodash.get(req, 'body.home-address-line-1', ''),
-    homeAddressLine2: lodash.get(req, 'body.home-address-line-2', ''),
-    homeAddressCity: lodash.get(req, 'body.home-address-city', ''),
-    homeAddressPostcode: lodash.get(req, 'body.home-address-postcode', ''),
-    dobDay: lodash.get(req, 'body.dob-day', ''),
-    dobMonth: lodash.get(req, 'body.dob-month', ''),
-    dobYear: lodash.get(req, 'body.dob-year', '')
+    firstName: lodash.get(req.body, FIRST_NAME_FIELD),
+    lastName: lodash.get(req.body, LAST_NAME_FIELD),
+    homeAddressLine1: lodash.get(req.body, HOME_ADDRESS_LINE1_FIELD),
+    homeAddressLine2: lodash.get(req.body, HOME_ADDRESS_LINE2_FIELD),
+    homeAddressCity: lodash.get(req.body, HOME_ADDRESS_CITY_FIELD),
+    homeAddressPostcode: lodash.get(req.body, HOME_ADDRESS_POSTCODE_FIELD),
+    dobDay: lodash.get(req.body, DOB_DAY_FIELD),
+    dobMonth: lodash.get(req.body, DOB_MONTH_FIELD),
+    dobYear: lodash.get(req.body, DOB_YEAR_FIELD),
   }
 
-  if (lodash.get(req, 'body.answers-checked') === 'true') {
+  if (!lodash.isEmpty(errors)) {
+    pageData['errors'] = errors
+    return response.response(req, res, 'stripe-setup/responsible-person/index', pageData)
+  } else if (lodash.get(req.body, 'answers-checked') === 'true') {
     return res.redirect(303, paths.dashboard.index)
-  } else if (lodash.get(req, 'body.answers-need-changing') === 'true') {
+  } else if (lodash.get(req.body, 'answers-need-changing') === 'true') {
     return response.response(req, res, 'stripe-setup/responsible-person/index', pageData)
+  } else {
+    return response.response(req, res, 'stripe-setup/responsible-person/check-your-answers', pageData)
   }
-  return response.response(req, res, 'stripe-setup/responsible-person/check-your-answers', pageData)
+}
+
+const validate = (req, fieldName, fieldValidator, maxLength) => {
+  const field = getFieldValue(req, fieldName)
```

Comment:
Is there a need for this method? It looks like it directly maps to `lodash.get`

Reply:
We’ll remove this in a future PR